src/cascadia/TerminalSettingsEditor/MainPage.cpp

```diff
@@ -144,4 +162,38 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
             _OpenJsonHandlers(nullptr, target);
         }
     }
+
+    void MainPage::_InitializeProfilesList()
+    {
+        // Manually create a NavigationViewItem for each profile
+        // and keep a reference to them in a map so that we
+        // can easily modify the correct one when the associated
+        // profile changes.
+        for (auto profile : _settingsSource.AllProfiles())
+        {
+            MUX::Controls::NavigationViewItem profileNavItem;
+            profileNavItem.Content(box_value(profile.Name()));
+            profileNavItem.Tag(box_value<guid>(profile.Guid()));
+
+            const auto iconSource{ IconPathConverter::IconSourceWUX(profile.Icon()) };
+            WUX::Controls::IconSourceElement icon;

```

Comment:
moderately horrified that the MUX nav view takes WUX icons...

Reply:
also did a quadruple take on this to make sure i wasn't going crazy 👀